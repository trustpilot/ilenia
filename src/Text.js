import withTranslations from './withTranslations';
import interpolate from './interpolate';
import PropTypes from 'prop-types';

const Text = ({ id, interpolations = {}, translations = {}, tag }) => {
  const string = translations[id];
  if (!string) {
    console.error(`Couldn't find '${id}' in the translations table`); // eslint-disable-line no-console
    return null;
  }

  return interpolate(string, interpolations, tag);
};

Text.propTypes = {
  id: PropTypes.string.isRequired,
  interpolations: PropTypes.objectOf(PropTypes.string),
  tag: PropTypes.shape({
    start: PropTypes.string,
    end: PropTypes.string,
  }),
};

export default withTranslations(Text);
