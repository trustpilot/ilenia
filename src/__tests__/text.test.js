import React from 'react';
import { render } from 'react-testing-library';
import LocalizationProvider from '../LocalizationProvider';
import Text from '../Text';

const translations = {
  test1: 'Just a random string',
  test2: 'A string but with a [value] in it',
  test3: 'A string that uses another {{placeholder}}',
};

const setupText = (string, interpolations, tag) => (
  <LocalizationProvider locale="en-US" translations={translations}>
    <Text id={string} interpolations={interpolations} tag={tag} translations={translations} />
  </LocalizationProvider>
);

test('Renders a plain string', () => {
  const { container } = render(setupText('test1'));

  expect(container.firstChild).toMatchSnapshot();
});

test('Renders a string with an interpolations', () => {
  const { container } = render(setupText('test2', { value: 'huge value' }));

  expect(container.firstChild).toMatchSnapshot();
});

test('Renders a string which uses a different placeholder syntax', () => {
  const { container } = render(
    setupText('test3', { placeholder: 'syntax' }, { start: '{{', end: '}}' })
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('Renders components if included as interpolations', () => {
  const { container } = render(
    setupText('test2', { value: <strong key="strong">huge value</strong> })
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('Renders nothing if the string is not there', () => {
  console.error = jest.fn(); // eslint-disable-line no-console

  const { container } = render(setupText('test4'));

  expect(container.firstChild).toMatchSnapshot();
  expect(console.error).toHaveBeenCalled(); // eslint-disable-line no-console
});
