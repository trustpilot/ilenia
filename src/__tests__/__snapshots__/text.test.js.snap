// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a plain string 1`] = `"Just a random string"`;

exports[`Renders a string which uses a different placeholder syntax 1`] = `"A string that uses another syntax"`;

exports[`Renders a string with an interpolations 1`] = `"A string but with a huge value in it"`;

exports[`Renders components if included as interpolations 1`] = `
Array [
  "A string but with a ",
  <strong>
    huge value
  </strong>,
  " in it",
]
`;

exports[`Renders nothing if the string is not there 1`] = `null`;
